beacon-chain/sync/error.go

```diff
@@ -91,11 +93,23 @@ func readStatusCodeNoDeadline(stream network.Stream, encoding encoder.NetworkEnc
 
 // only returns true for errors that are valid (no resets or expectedEOF errors).
 func isValidStreamError(err error) bool {
-	return err != nil && !errors.Is(err, mux.ErrReset) && !errors.Is(err, helpers.ErrExpectedEOF)
+	return err != nil && !errors.Is(err, mux.ErrReset)
 }
 
 func closeStream(stream network.Stream, log *logrus.Entry) {
-	if err := helpers.FullClose(stream); err != nil && err.Error() != mux.ErrReset.Error() {
-		log.WithError(err).Debug("Could not reset stream")
+	if err := stream.Close(); isValidStreamError(err) {
+		log.WithError(err).Debugf("Could not reset stream with protocol %s", stream.Protocol())
 	}
 }
+
+func closeStreamAndWait(stream network.Stream, log *logrus.Entry) {
+	if err := stream.CloseWrite(); err != nil {
+		stream.Reset()
```

Comment:
Please handle this error from stream.Reset(). (Static analysis will not allow the build to complete with unhandled errors)